ADD Raygun integrations, env-var snapshot functionality

dnsl48 opened the pull request:
Resolves:
 - https://github.com/silverstripeltd/open-sourcerers/issues/64

Requires:
 - https://github.com/silverstripe/silverstripe-raygun/pull/52
 - https://github.com/sminnee/silverstripe-tagmanager/pull/11

dnsl48 committed:
ADD Raygun integrations, env-var snapshot functionality

unclecheese reviewed (commented):
Really clever solution. I like it!

My only question was whether `flush` is the most appropriate hook for refreshing the variables? Since changes to environment variables don't require flushing as is, this would be an entirely new pattern anyway, and I'm wondering, since its job is to write to the database, if it would be more appropriate to put this in the `dev/build` cycle, e.g. `requireDefaultRecords()`.

dnsl48 commented:
Yes, sounds like `requireDefaultRecords` might be more appropriate as it needs to write to the database. I guess I only chose `?flush` because I'm more familiar with it. On the other hand, I'm not sure how idempotent `requireDefaultRecords` is implied to be? Because ideally we should be able to update the variables when we update them in the platform UI, whereas require default records sounds like a one-off operation?

unclecheese commented:
You'd think so, but it actually runs every build. It's ~~weird~~ poorly named.

dnsl48 commented:
Fixed. Now it's on `requireDefaultRecords`. I also added removing for empty values, otherwise we won't have any means to delete variables from the database.

Commit:
Rebase, minor refactoring

Commit:
Linting

unclecheese commented:
@dnsl48 Have done some minor refactoring. If looks okay to you, we can merge.

dnsl48 commented:
yes, looks good, thank you. Though, we may need to merge the patch to tagmanager first, because currently composer.json in this PR looks at open-sausages/tagmanager

Commit:
Rebase, composer update

Commit:
Rebase, composer update, new alteration messages

Commit:
Composer update to latest raygun

chillu commented on public/index.php:
```diff
@@ -22,6 +22,6 @@
 // Default application
 $kernel = new CoreKernel(BASE_PATH);
 $app = new HTTPApplication($kernel);
-$app->addMiddleware(new ErrorControlChainMiddleware($app));
```
@dnsl48 Why are you *replacing* error control chain here? Your middleware does something quite different, shouldn't it be added instead?

dnsl48 replied:
we only have ErrorControlChainMiddleware in the first place here because we forgot to delete it from the recipe-core when we released 4.4

chillu replied:
lol fair enough.
